save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ContestExecutionView.kt

```diff
@@ -0,0 +1,250 @@
+                        }
+                    }
+                }
+            }
+            column("endDate", "End time", { endTime }) { cellProps ->
+                Fragment.create {
+                    td {
+                        a {
+                            +(formattingDate(cellProps.value) ?: "Starting")
+                        }
+                    }
+                }
+            }
+            column("running", "Running", { runningTests }) { cellProps ->
+                Fragment.create {
+                    td {
+                        a {
+                            +"${cellProps.value}"
+                        }
+                    }
+                }
+            }
+            column("passed", "Passed", { passedTests }) { cellProps ->
+                Fragment.create {
+                    td {
+                        a {
+                            +"${cellProps.value}"
+                        }
+                    }
+                }
+            }
+            column("failed", "Failed", { failedTests }) { cellProps ->
+                Fragment.create {
+                    td {
+                        a {
+                            +"${cellProps.value}"
+                        }
+                    }
+                }
+            }
+            column("skipped", "Skipped", { skippedTests }) { cellProps ->
+                Fragment.create {
+                    td {
+                        a {
+                            +"${cellProps.value}"
+                        }
+                    }
+                }
+            }
+        },
+        getRowProps = { row ->
+            val color = when (row.original.status) {
+                ExecutionStatus.ERROR -> Colors.RED
+                ExecutionStatus.PENDING -> Colors.GREY
+                ExecutionStatus.RUNNING -> if (row.original.failedTests != 0L) Colors.DARK_RED else Colors.GREY
+                ExecutionStatus.FINISHED -> if (row.original.failedTests != 0L) Colors.DARK_RED else Colors.GREEN
+            }
+            jso {
+                style = jso {
+                    background = color.value.unsafeCast<Background>()
+                }
+            }
+        },
+        renderExpandedRow = { tableInstance, row ->
+            tr {
+                td {
+                    colSpan = tableInstance.columns.size
+                    div {
+                        className = ClassName("row")
+                        displayExecutionInfoHeader(row.original, "row col-11")
+                        div {
+                            className = ClassName("col-1")
+                            pieChart(
+                                getPieChartData(row.original),
+                            ) { pieProps ->
+                                pieProps.animate = true
+                                pieProps.segmentsShift = 2
+                                pieProps.radius = 47
```

Comment:
Should these be dependent on window size or screen resolution?

Reply:
Seems that radius is set in percents of parent minimal dimension